components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -125,9 +132,47 @@ void DeletePromoCodeFile(const base::FilePath& promo_code_file) {
   }
 }
 
+#if BUILDFLAG(IS_MAC)
+std::string ReadPromoCodeFromXattr() {
+  static const char kRefCodeAttr[] = "com.brave.refcode";
+  base::FilePath bundle_path = base::apple::OuterBundlePath();
+  std::vector<char> value;
+  int result_errno =
+      brave::GetFileExtendedAttribute(bundle_path, kRefCodeAttr, &value);
+  if (result_errno == ENOATTR) {
+    LOG(WARNING) << "Could not get promo code from " << bundle_path
+                 << ". The extended attribute " << kRefCodeAttr
+                 << " was not found.";
+  } else if (result_errno != 0) {
+    LOG(ERROR) << "Could not get promo code from " << bundle_path
+               << ". An error occurred getting value for attribute "
+               << kRefCodeAttr << ". Error code: " << result_errno << ".";
+  } else {
+    std::string promo_code(value.begin(), value.end());
+    base::TrimWhitespaceASCII(promo_code, base::TRIM_ALL, &promo_code);
+    if (promo_code.empty()) {
+      LOG(ERROR) << "Promo code value from " << bundle_path
+                 << "'s extended attribute " << kRefCodeAttr << " is empty.";
+    } else {
+      DVLOG(1) << "Promo code from " << kRefCodeAttr << ": " << promo_code;
+      return promo_code;
+    }
+  }
+
+  return "";
```

Comment:
nit: `base::EmptyString();`

Reply:
From the upstream comment it seems that we shouldn't use `base::EmptyString()` in this case (https://source.chromium.org/chromium/chromium/src/+/main:base/strings/string_util.h;l=204): 

> ...There is only one case where you should use these: functions
// which need to return a string by reference (e.g. as a class member
// accessor), and don't have an empty string to use (e.g. in an error case).
// These should not be used as initializers, function arguments, or return
// values for functions which return by value or outparam.

Reply:
Got it! Thanks for checking 👍🏼